packages/thirdweb/src/react/web/ui/prebuilt/thirdweb/ClaimButton/index.tsx

```diff
@@ -0,0 +1,195 @@
+            const [is721, claimSupported] = await Promise.all([
+              isERC721({ contract }),
+              isClaimSupported(contract),
+            ]);
+            if (!is721 || !claimSupported) {
+              throw new Error("Not a valid NFT Drop (ERC721) contract");
+            }
+            return claimTo({
+              contract,
+              to: account?.address || "",
+              quantity: claimParams.quantity,
+            });
+          }
+          case "ERC1155": {
+            const [{ isERC1155 }, { isClaimSupported }, { claimTo }] =
+              await Promise.all([
+                import(
+                  "../../../../../../extensions/erc1155/read/isERC1155.js"
+                ),
+                import(
+                  "../../../../../../extensions/erc1155/__generated__/IDrop1155/write/claim.js"
+                ),
+                import(
+                  "../../../../../../extensions/erc1155/drops/write/claimTo.js"
+                ),
+              ]);
+            const [is721, claimSupported] = await Promise.all([
+              isERC1155({ contract }),
+              isClaimSupported(contract),
+            ]);
+            if (!is721 || !claimSupported) {
+              throw new Error("Not a valid Edition Drop (ERC1155) contract");
+            }
+            return claimTo({
+              contract,
+              to: account?.address || "",
+              quantity: claimParams.quantity,
+              tokenId: claimParams.tokenId,
+            });
+          }
+          case "ERC20": {
+            const [{ isERC20 }, { isClaimSupported }, { claimTo }] =
+              await Promise.all([
+                import("../../../../../../extensions/erc20/read/isERC20.js"),
+                import(
+                  "../../../../../../extensions/erc20/__generated__/IDropERC20/write/claim.js"
+                ),
+                import(
+                  "../../../../../../extensions/erc20/drops/write/claimTo.js"
+                ),
+              ]);
+            const [is721, claimSupported] = await Promise.all([
+              isERC20({ contract }),
+              isClaimSupported(contract),
+            ]);
+            if (!is721 || !claimSupported) {
+              throw new Error("Not a valid Token Drop (ERC20) contract");
+            }
+            if ("quantity" in claimParams) {
+              return claimTo({
+                contract,
+                to: account?.address || "",
+                quantity: claimParams.quantity,
+              });
+            }
+            if ("quantityInWei" in claimParams) {
+              return claimTo({
+                contract,
+                to: account?.address || "",
+                quantityInWei: claimParams.quantityInWei,
+              });
+            }
+            throw new Error("Missing quantity or quantityInWei");
+          }
+          default:
+            throw new Error("Invalid contract type. Must be ERC721/1155/20");
```

Comment:
can improve this error message, at least ERC<x> for each of them instead of `/` between the numbers